### DISCUSSION TITLE
Optimizer

### DISCUSSION THREAD

**Discussion by swissdigitalnet** [wlcrs/huawei_solar#190] [Setting up the integration]
I am happy with this piece of software. Thank you! During installation, I di not check the "Elevated Permissions" because I did not want to change settings.
Now I learned I should have checked it because the optimizer data is unavailable. I tried to set up the integration again, but without success. I even had to restore a backup to make it work again.
What is the right procedure to enable this feature after the initial setup?

**Comment by SensorsIot** [comment]
Sorry, I used the wrong account. This is me.

**Comment by stefanhapper** [comment]
> What is the right procedure to enable this feature after the initial setup?

I have the same question. I can off course re-install but I wam worried that I will lose my data.

**Comment by wlcrs** [comment]
The correct procedure is to remove the existing integration, and add it again. As the entities will be generated again with the identical name they had earlier, their history is preserved. (At least, this is what I have observed myself during development and the many re-installs that are done because of this).

If this does not work for you, you are always welcome to submit a bugreport with all the logging (including pyModbus) enabled showing the issue.

A suggestion: have you tried restarting HA between removing the integration and adding it back again? (should not be necessary, but makes sure that any lingering process is stopped)
